### PR TITLE
Finish env overrides for DB commands

### PR DESCRIPTION
**Pull request by samtstern**
### Description

This finishes adding support for `FIRESTORE_EMULATOR_HOST` and `FIREBASE_DATABASE_EMULATOR_HOST` wherever reasonable and adds warnings/prompts to all other commands that may confuse developers.

TODO:

- [x] Finish documenting new `utils` methods
- [x] Unit tests

### Scenarios Tested

**Database Set and Get**

These operations are supported.

```bash
$ FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000 firebase database:set /foo/bar data.json
i  You have set FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000, this command will execute against the Realtime Database emulator running at that address.
? You are about to overwrite all data at http://localhost:9000/foo/bar?ns=fir-dumpster. Are you sure? Yes

✔  Data persisted successfully

View data at: http://localhost:9000/foo/bar.json?ns=fir-dumpster
```

```-
$ FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000 firebase database:get /foo/bar
i  You have set FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000, this command will execute against the Realtime Database emulator running at that address.
{"hello":"world"}
```

**Database Remove**

This operation is not supported.

```bash
$ FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000 firebase database:remove /foo/list
⚠  You have set FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000, however this command does not support running against the emulator
 so this action will affect production.
? Do you want to continue? No

Error: Command aborted.
```

### CODE REVIEW

File: src/emulator/commandUtils.ts [hunk 1]
```diff
@@ -45,27 +46,81 @@ export const DESC_TEST_PARAMS =
  */
 const DEFAULT_CONFIG = new Config({ database: {}, firestore: {}, functions: {}, hosting: {} }, {});
 
-export function warnRealtimeDatabaseEmulated(): void {
+export function warnRealtimeDatabaseEmulated(
```
Comment (yuchenshi) [inline src/emulator/commandUtils.ts:49]: Please split this into two functions, `printNoticeIfEmulated` and `confirmAffectingProd` so it is much easier to read in `commands/*.ts`. Right now, it is super hard to follow what `.before(warnRealtimeDatabaseEmulated, false)` actually means and what it does.

nit: you can reuse logic across RTDB and Firestore like `.before(printNoticeIfEmulated, Emulators.DATABASE)`. In the implementation, use `Constants.description(emulator)` in `src/emulators/constants.ts` to construct the log messages. A constant map from the `Emulators` enum to the env var controlling it would also be nice.

Comment (samtstern) [reply src/emulator/commandUtils.ts]: Change made, thank you this was a great idea and it's much better now.

File: src/emulator/commandUtils.ts [hunk 2]
```diff
@@ -46,81 +46,60 @@ export const DESC_TEST_PARAMS =
  */
 const DEFAULT_CONFIG = new Config({ database: {}, firestore: {}, functions: {}, hosting: {} }, {});
 
-export function warnRealtimeDatabaseEmulated(
-  options: any,
-  emulatorSupported: boolean
-): void | Promise<void> {
-  const envKey = Constants.FIREBASE_DATABASE_EMULATOR_HOST;
+export function printNoticeIfEmulated(options: any, emulator: Emulators): void {
```
Comment (yuchenshi) [inline src/emulator/commandUtils.ts:49]:
```suggestion
export function printNoticeIfEmulated(options: any, emulator: Emulators.DATABASE | Emulators.FIRESTORE): void {
```

Right now this function only supports those two emulators so let's just be frank about it. Right now, you can pass in `Emulators.HOSTING` and it actually checks the Firestore env var, which is super bad.

File: src/emulator/commandUtils.ts [hunk 3]
```diff
@@ -45,27 +46,60 @@ export const DESC_TEST_PARAMS =
...
+      ? Constants.FIREBASE_DATABASE_EMULATOR_HOST
+      : Constants.FIRESTORE_EMULATOR_HOST;
   const envVal = process.env[envKey];
+
   if (envVal) {
     utils.logWarning(
       `You have set ${clc.bold(
         `${envKey}=${envVal}`
-      )}, this command will execute against the Cloud Firestore emulator running at that address.`
+      )}, however this command does not support running against the ${emuName} emulator so this action will affect production.`
```
Comment (yuchenshi) [inline src/emulator/commandUtils.ts:85]:
```suggestion
      )}, however this command does not support running against the ${emuName} so this action will affect production.`
```

File: src/emulator/commandUtils.ts [hunk 4]
```diff
@@ -45,27 +46,60 @@ export const DESC_TEST_PARAMS =
...
+      ? Constants.FIREBASE_DATABASE_EMULATOR_HOST
+      : Constants.FIRESTORE_EMULATOR_HOST;
   const envVal = process.env[envKey];
   if (envVal) {
-    utils.logWarning(
+    utils.logBullet(
       `You have set ${clc.bold(
         `${envKey}=${envVal}`
-      )}, this command will execute against the Realtime Database emulator running at that address.`
+      )}, this command will execute against the ${emuName} emulator running at that address.`
```
Comment (yuchenshi) [inline src/emulator/commandUtils.ts:64]:
```suggestion
      )}, this command will execute against the ${emuName} running at that address.`
```
